polyfill/lib/casts.mjs

```diff
@@ -143,14 +143,11 @@ export function CastTime(arg, aux) {
 
 export function CastYearMonth(arg, aux) {
   const YearMonth = ES.GetIntrinsic('%Temporal.YearMonth%');
-  if (HasSlot(arg, YEAR, MONTH)) {
-    if (!HasSlot(arg, DAY)) return arg;
-    return new YearMonth(GetSlot(arg, year), GetSlot(arg, MONTH));
+  if (HasSlot(arg, YEAR, MONTH) && !HasSlot(arg, DAY)) {
+    return arg;
   }
   if ('string' === typeof arg) {
```

Comment:
nbd but why the yoda condition?

Reply:
@pipobscure

Reply:
I don't understand the question: because it allows `TemporalObject.from(isostring)` to be valid. This is very useful for making `.from` be the universal "try hard to cast this" operation.

Reply:
I think this was a very minor stylistic question, and we don't need to follow up further.

Reply:
It was stylistic and minor; to clarify, I’m asking why “string equals arg” instead of the more common “arg equals string”. Colloquially, putting the variable last is called a “yoda condition”.